### ISSUE TITLE
[JOSS review] [bug?] Submitting only one `lapply` job fails

### ISSUE THREAD

**Issue by mschubert** [USCbiostats/sluRm#12]
This is a part of the JOSS review outlined in https://github.com/openjournals/joss-reviews/issues/1493.

The following fails:

```r
library(sluRm)
Slurm_lapply(1, function(x) x*2)
# Error in splits[[j]] : subscript out of bounds
# In addition: Warning message:
# `X` is not a list. The function will coerce it into one using `as.list`
```

Related: Internal parallelism via `mc.cores` defaults to `2`, which wastes a core if there are not more calls than jobs.

**Comment by gvegayon** [comment]
Thanks. This should be fixed now. Regarding the default of 2 mc.cores, that's not an issue since `mclapply` automatically checks for this. If length(X) == 1, then mc.cores defaults to `lapply` instead. I was just trying to be consistent with what is in `parallel`.

**Comment by gvegayon** [comment]
Also, this was a nasty bug. Thanks for flagging that!

**Comment by mschubert** [comment]
> Regarding the default of 2 mc.cores, that's not an issue since `mclapply` automatically checks for this. If length(X) == 1, then mc.cores defaults to `lapply` instead.

That's true for mclapply in R, but not for the job that requests 2 cores (and only uses 1)

**Comment by gvegayon** [comment]
OK, good point. Nevertheless, I'm honestly not sure regarding this as a
problem, since in most cases using Slurm_lapply/Slurm_Map/Slurm_sapply
users will be submitting a job with length(X) >> 1. I do need to get more
details about what happens behind scenes when you request more than you are
using, and also regarding how much enforced is the request of resources (in
the manual of sbatch, the "Advice the controller" phrase is a bit confusing
for me).

Either way, I follow your advice and will set the default to equal 1L.

[ ] Set `mc.cores = 1L` in `Slurm_lapply`/`Map`

George G. Vega Yon
+1 (626) 381 8171
https://ggvy.cl

**Comment by gvegayon** [comment]
Now in 91909a7c4fd4b94811d473384b56712285e7baa5
